src/components/validation/validators/component_spec/sources.rs

```diff
@@ -85,22 +112,7 @@ fn validate_component_received_events_total(
         TEST_SOURCE_NAME,
     )?;
 
-    let mut events = 0;
-    for m in metrics {
-        match m.value() {
-            vector_core::event::MetricValue::Counter { value } => {
-                if let MetricKind::Absolute = m.data().kind {
-                    events = *value as i32
-                } else {
-                    events += *value as i32
-                }
-            }
-            _ => errs.push(format!(
-                "{}: metric value is not a counter",
-                SourceMetrics::EventsReceived,
-            )),
-        }
-    }
+    let events: i32 = sum_counters(SourceMetrics::EventsReceived, &metrics)? as i32;
```

Comment:
ok, it's just a suggestion. You could still keep the `sum` an f64 as you sum up the values, since that is the value type of `MetricValue::Counter` ,  and then use the generic to do the casting within the helper function, to make the call sites a little cleaner looking, as it replaces the explicit type declaration with a generic parameter, and avoids the cast.
Up to you though, this is a pedantic thing.

Reply:
I removed the usages of the unnecessary cast of f64 -> f64.
I'm fine dropping this suggestion.